ocaml/xapi/test_vdi_cbt.ml

```diff
@@ -209,6 +223,58 @@ let test_allowed_operations_updated_when_necessary () =
   Client.Client.VDI.data_destroy ~rpc ~session_id ~self;
   assert_allowed_operations "does not contain `copy after VDI has been data-destroyed" (fun ops -> not @@ List.mem `copy ops)
 
+(* Confirm VDI.data_destroy changes requisite fields of VDI *)
+let test_vdi_after_data_destroy () =
+  try
+    let __context = Test_common.make_test_database () in
+
+    (* Create host -> (SM) -> (PBD) -> SR -> VDI -> (VBD) -> VM infrastructure
+       in order to run VDI.data_destroy and test suspended VMs *)
+    let host = Helpers.get_localhost ~__context in
+    let _: _ API.Ref.t = Test_common.make_sm ~__context () in
+    let sR = Test_common.make_sr ~__context ~_type:"sm" ~is_tools_sr:false () in
+    let _: _ API.Ref.t = Test_common.make_pbd ~__context ~host ~sR ~currently_attached:true () in
+    let vDI = Test_common.make_vdi ~__context ~is_a_snapshot:true ~managed:true ~cbt_enabled:true ~_type:`suspend ~sR () in
+    let vM = Test_common.make_vm ~__context () in
+    let _: _ API.Ref.t = Test_common.make_vbd ~__context ~uuid:"VBD-1" ~vDI ~vM ~currently_attached:false () in
+    register_smapiv2_server ~vdi_data_destroy:(fun _ ~dbg ~sr ~vdi -> ()) (Db.SR.get_uuid ~__context ~self:sR);
+
+    let check_vdi_is_snapshot_and_type ~vDI ~snapshot ~vdi_type =
```

Comment:
Since we always call this with `~snapshot:true`, we could remove this param and hardcode it.